module/zfs/zfs_ioctl.c

```diff
@@ -3447,11 +3439,8 @@ static const zfs_ioc_key_t zfs_keys_remap[] = {
 static int
 zfs_ioc_remap(const char *fsname, nvlist_t *innvl, nvlist_t *outnvl)
 {
-	if (strchr(fsname, '@') ||
-	    strchr(fsname, '%'))
-		return (SET_ERROR(EINVAL));
-
-	return (dmu_objset_remap_indirects(fsname));
+	/* This IOCTL is no longer supported. */
+	return (0);
```

Comment:
Rather than returning success shouldn't we be returning `EOPNOTSUPP` to indicate this ioctl is no longer supported.

Reply:
I thought that ENOTSUPP means specifically that the pool needs to be upgraded?

That said I could be convinced to make this some sort of error rather than silent success.  For this IOCTL it seems OK either way since it was always "best effort" (i.e. "please optimize stuff as best as you can", and now we're saying "it's already great, nothing to do here").

Reply:
The ENOTSUPP error message is interpretted by userspace as "pool must be upgraded". Perhaps, this would be a place where another custom zfs errno might make sense?

Reply:
For this IOCTL I think we're OK letting it return success like this, since as you pointed out it was always best effort anyway.  If we do remove something more critical in the future I think we'll have to reserve some errno for it.